File: spec/system/facets_spec.rb
require "rails_helper"

describe "Facets", js: true do
  it "displays facets in a specific order" do
    add_document('bloch.xml')
    add_document('intelligencer.xml')
    add_document('kopit.xml')
    add_document('oh002.xml')

    visit "/"

    within("div#facet-panel-collapse") do
      facets = page.all(".facet-limit").map(&:text)

      expect(facets).to eq [
                             "Library",
                             "Digital Content",
                             "Creator",
                             "Date Range",
                             "Subject",
                             "Name",
                             "Place",
                             "Language",
                             "Collection",
                             "Level"
                           ]
    end
  end
end

File: e2e/tests/facets.spec.js
import { expect, test } from '@playwright/test';

const EXPECTED_FACET_ORDER = [
  'Library',
  'Digital Content',
  'Creator',
  'Date Range',
  'Subject',
  'Name',
  'Place',
  'Language',
  'Collection',
  'Level',
];

test.describe('Facets', () => {
  test('displays facets in a specific order', async ({ page }) => {
    await page.goto('/');

    const facetButtons = page.locator('#facet-panel-collapse .facet-field-heading button');

    await expect(facetButtons).toHaveCount(EXPECTED_FACET_ORDER.length);

    const actualOrder = await facetButtons.allTextContents();
    const normalizedOrder = actualOrder.map((text) => text.replace(/\s+/g, ' ').trim());

    expect(normalizedOrder).toEqual(EXPECTED_FACET_ORDER);
  });
});
